stdlib/public/core/Integers.swift.gyb

```diff
@@ -3232,257 +3183,496 @@ public struct DoubleWidth<T : FixedWidthInteger>
   // Numeric
   //
   public init() {
-    self.init((High(), Low()))
+    self.init((0, 0))
   }
 
   // integer
   //
   public var magnitude: DoubleWidth<Low> {
-    if T.isSigned && _storage.high < 0 {
-        return (DoubleWidth<T>() - self).magnitude
+    if Base.isSigned && _storage.high < 0 {
+      return self == .min
+        ? DoubleWidth.max.magnitude &+ 1
+        : (0 - self).magnitude
     }
     return DoubleWidth<Low>((
       _storage.high.magnitude, _storage.low.magnitude))
   }
 
-  public func isEqual(to rhs: DoubleWidth<T>) -> Bool {
-    return (_storage.high == rhs._storage.high) &&
-           (_storage.low == rhs._storage.low)
+  internal init(_ _magnitude: Magnitude) {
+    self.init((High(_magnitude._storage.high), _magnitude._storage.low))
   }
 
-  public func isLess(than rhs: DoubleWidth<T>) -> Bool {
-    if _storage.high < rhs._storage.high {
+  public static func ==(lhs: DoubleWidth, rhs: DoubleWidth) -> Bool {
+    return (lhs._storage.high == rhs._storage.high) &&
+      (lhs._storage.low == rhs._storage.low)
+  }
+
+  public static func <(lhs: DoubleWidth, rhs: DoubleWidth) -> Bool {
+    if lhs._storage.high < rhs._storage.high {
       return true
     }
-    if (_storage.high > rhs._storage.high) {
+    if lhs._storage.high > rhs._storage.high {
       return false
     }
-    return _storage.low < rhs._storage.low
+    return lhs._storage.low < rhs._storage.low
   }
-
+    
   public init<T : BinaryInteger>(_ source: T) {
-    fatalError()
+    self.init(exactly: source)!
   }
 
   public init?<T : BinaryInteger>(exactly source: T) {
-    fatalError()
-  }
-
-  public init<T : BinaryInteger>(extendingOrTruncating source: T) {
-    fatalError()
+    // Can't represent a negative 'source' if Base is unsigned
+    guard source >= 0 || DoubleWidth<Base>.isSigned else { return nil }
```

Comment:
Nobody has to write `(0 as T)`. It will work without an explicit type context just fine, unless you're messing with the guts of integers, where incorrect overload can lead to infinite recursion.